common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCache.java

```diff
@@ -113,15 +152,25 @@ public synchronized void saveAccount(@NonNull final AccountRecord accountToSave)
         }
 
         final String cacheValue = mCacheValueDelegate.generateCacheValue(accountToSave);
-        mSharedPreferencesFileManager.put(cacheKey, cacheValue);
+        mLock.writeLock().lock();
+        try {
+            clearAccountAndCredentialsList();
```

Comment:
We're clearning the list everytime we save an account?

Reply:
Yeah that's a good idea. Found an implementation which is kind of doing the same thing that I am introducing in this PR. I will move to using that implementation behind a feature flag.